llama_stack/apis/safety/safety.py

```diff
@@ -38,15 +38,10 @@ class RunShieldResponse(BaseModel):
     violation: Optional[SafetyViolation] = None
 
 
-class ShieldStore(Protocol):
-    async def get_shield(self, identifier: str) -> ShieldDef: ...
-
-
 @runtime_checkable
 class Safety(Protocol):
-    shield_store: ShieldStore
 
     @webmethod(route="/safety/run_shield")
     async def run_shield(
-        self, identifier: str, messages: List[Message], params: Dict[str, Any] = None
+        self, shield: Shield, messages: List[Message], params: Dict[str, Any] = None
```

Comment:
I think we decided that we will only pass in what @raghotham called the "key" -- i.e., the ID which the user associated with the object. The data model for `Resource` needs to be updated too. Basically it should have three important parts (other than type):

- "identifier"  --> this is identity of the object in the provider. only the provider can make sense of this name and is necessary. e.g., `fireworks/llama-3pv2-vision-instruct`
- "name" or "key" --> this is the name the user decides to call this object for the purposes of llama stack. e.g., "vision_model_for_furniture" 
- "provider_id"  --> non-controversial

now I don't think the terms "name"/"key" and "identifier" are completely self-evident from the purposes of a user. maybe calling the first "alias" would be better? 

cc @raghotham for thoughts here.

Reply:
updated.